src/clj/com/helpshift/zk_queue.clj

```diff
@@ -65,6 +70,20 @@
                     {:nodes zk-nodes}))))
 
 
+
+(defn init-queue-with-zk-client
+  "Give queue-path and zookeeper client.
+   Returns queue object"
+  [zk-client zk-queue-path]
+  (assert (instance? ZooKeeper zk-client)
```

Comment:
Why don't we put these asserts in preconditions? Is it because we need a custom error string?